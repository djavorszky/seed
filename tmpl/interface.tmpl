package {{ .PackageName }}

{{ range .Imports }}import "{{.}}"{{ end }}

// {{.ServiceName}}Service encapsulates the handler interface, which holds all the methods to be called
// by the server, and middleware interface, which contains all the middlewares to be added to the service.
type {{.ServiceName}}Service interface {
	{{.ServiceName}}Handler
	{{.ServiceName}}Middleware
}

// {{.ServiceName}}Handler is the interface for the handlers. Any new endpoint added by seed will be added here as a
// new method on the interface.
type {{.ServiceName}}Handler interface {
    {{ range .Services}}{{.}}{{ end }}
}

// {{.ServiceName}}Middleware is the interface for all the middlewares that will be added to all of the paths.
type {{.ServiceName}}Middleware interface {
    {{ range .Middlewares}}{{.}}{{ end }}
}
