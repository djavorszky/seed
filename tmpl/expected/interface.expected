package gen

import "net/http"

// AdmiralService encapsulates the handler interface, which holds all the methods to be called
// by the server, and middleware interface, which contains all the middlewares to be added to the service.
type AdmiralService interface {
	AdmiralHandler
	AdmiralMiddleware
}

// AdmiralHandler is the interface for the handlers. Any new endpoint added by seed will be added here as a
// new method on the interface.
type AdmiralHandler interface {
	Index() http.HandlerFunc
}

// AdmiralMiddleware is the interface for all the middlewares that will be added to all of the paths.
type AdmiralMiddleware interface {
	LoggerMw(http.Handler) http.Handler
}
